experiments/yarn/src/main/scala/org/apache/gearpump/experiments/yarn/master/AmActor.scala

```diff
@@ -18,69 +18,113 @@
 
 package org.apache.gearpump.experiments.yarn.master
 
-import java.io.File
 import java.net.InetAddress
 import java.util.concurrent.TimeUnit
 
+import akka.actor.{Actor, ActorRef, ActorSystem, PoisonPill, Props, actorRef2Scala}
+import akka.util.Timeout
+import com.typesafe.config.ConfigFactory
 import org.apache.gearpump.cluster.main.{ArgumentsParser, CLIOption}
-import org.apache.gearpump.experiments.yarn.Actions.{AMStatusMessage, AllRequestedContainersCompleted, ContainerInfo, ContainerRequestMessage, Failed, LaunchContainers, RMHandlerDone, RegisterAMMessage, ShutdownRequest, _}
 import org.apache.gearpump.experiments.yarn.CmdLineVars.{APPMASTER_IP, APPMASTER_PORT}
 import org.apache.gearpump.experiments.yarn.Constants._
-import org.apache.gearpump.experiments.yarn.{AppConfig, NodeManagerCallbackHandler, ResourceManagerClientActor}
+import org.apache.gearpump.experiments.yarn._
+import org.apache.gearpump.experiments.yarn.master.AmActor.{RMCallbackHandlerActorProps, RMClientActorProps}
+import org.apache.gearpump.transport.HostPort
 import org.apache.gearpump.util.LogUtil
 import org.apache.hadoop.conf.Configuration
 import org.apache.hadoop.net.NetUtils
-import org.apache.hadoop.yarn.api.ApplicationConstants
 import org.apache.hadoop.yarn.api.protocolrecords.RegisterApplicationMasterResponse
 import org.apache.hadoop.yarn.api.records.{Container, ContainerId, FinalApplicationStatus}
 import org.apache.hadoop.yarn.client.api.async.NMClientAsync
-import org.apache.hadoop.yarn.client.api.async.impl.NMClientAsyncImpl
 import org.apache.hadoop.yarn.conf.YarnConfiguration
 import org.slf4j.Logger
-import com.typesafe.config.{ConfigParseOptions, ConfigFactory}
-import akka.actor.Actor
-import akka.actor.ActorRef
-import akka.actor.ActorSystem
-import akka.actor.PoisonPill
-import akka.actor.Props
-import akka.actor.actorRef2Scala
-import akka.util.Timeout
-import org.apache.gearpump.experiments.yarn.Actions._
-import org.apache.gearpump.transport.HostPort
 
-import org.apache.gearpump.util.Constants
+import scala.util.{Failure, Success, Try}
 
+object AmActor {
+  case class RMCallbackHandlerActorProps(props: Props)
+  case class RMClientActorProps(props: Props)
+
+  def getRMCallbackHandlerActorProps(appConfig: AppConfig) : RMCallbackHandlerActorProps =  RMCallbackHandlerActorProps(Props(classOf[RMCallbackHandlerActor], appConfig))
+  def getRMClientActorProps(yarnConfiguration: YarnConfiguration) : RMClientActorProps =  RMClientActorProps(Props(classOf[ResourceManagerClientActor], yarnConfiguration))
+}
+
+object AmStates {
+  sealed trait Reason
+  case class Failed(throwable: Throwable) extends Reason
+  case object ShutdownRequest extends Reason
+  case object AllRequestedContainersCompleted extends Reason
+}
+
+object AmActorProtocol {
```

Comment:
As message here is part of the interface, may  be we should group them and document them better?